admission-policy/readme.adoc

```diff
@@ -0,0 +1,342 @@
+kubectl create -f opa-admission-controller-service.yaml -n opa
+```
+
+Next, create Secrets containing the TLS credentials for OPA:
+
+```bash
+kubectl create secret generic opa-ca --from-file=ca.crt -n opa
+kubectl create secret tls opa-server --cert=server.crt --key=server.key -n opa
+```
+
+Finally, create the Deployment to run OPA as an Admission Controller.
+
+**opa-admission-controller-deployment.yaml**:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: opa
+  name: opa
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: opa
+      name: opa
+    spec:
+      containers:
+        - name: opa
+          image: openpolicyagent/opa:0.5.13
+          args:
+            - "run"
+            - "--server"
+            - "--tls-cert-file=/certs/tls.crt"
+            - "--tls-private-key-file=/certs/tls.key"
+            - "--addr=0.0.0.0:443"
+            - "--insecure-addr=127.0.0.1:8181"
+          volumeMounts:
+            - readOnly: true
+              mountPath: /certs
+              name: opa-server
+        - name: kube-mgmt
+          image: openpolicyagent/kube-mgmt:0.5
+          args:
+            - "--replicate=v1/pods"
+            - "--pod-name=$(MY_POD_NAME)"
+            - "--pod-namespace=$(MY_POD_NAMESPACE)"
+            - "--register-admission-controller"
+            - "--admission-controller-ca-cert-file=/certs/ca.crt"
+            - "--admission-controller-service-name=opa"
+            - "--admission-controller-service-namespace=$(MY_POD_NAMESPACE)"
+          volumeMounts:
+            - readOnly: true
+              mountPath: /certs
+              name: opa-ca
+          env:
+            - name: MY_POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: MY_POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+      volumes:
+        - name: opa-server
+          secret:
+            secretName: opa-server
+        - name: opa-ca
+          secret:
+            secretName: opa-ca
+```
+
+```bash
+kubectl create -f opa-admission-controller-deployment.yaml -n opa
+```
+
+When OPA starts, the sidecar (`kube-mgmt`) will register it as an External
+Admission Controller. To verify that registration succeeded, run `kubectl proxy`
+in another terminal and then query the Kubernetes API for the list of External
+Admission Controllers.
+
+```bash
+curl localhost:8001/apis/admissionregistration.k8s.io/v1alpha1/externaladmissionhookconfigurations
+```
+
+Finally, you can follow the OPA logs to see the webhook requests being issued
+by the Kubernetes API server:
+
+```
+kubectl logs -l app=opa -c opa -n opa
+```
+
+== Step 3:  Enforce Kubernetes Admission Control with OPA
+
+=== 3.1 Load a policy into OPA
+To test admission control, create a policy that restricts exec access on
+privileged pods:
+
+**image_source.rego**:
+
+```ruby
```

Comment:
Can these policies be only written in Ruby? Provide a link to how more extensive policies can be written?

Reply:
Added a link to policy language docs.  The language isn't Ruby; that's just the closest syntax highlighter that we've found.